source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -396,8 +396,8 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
   const auto& filters = config.http_filters();
   for (int32_t i = 0; i < filters.size(); i++) {
     bool is_terminal = false;
-    processFilter(filters[i], i, "http", filter_factories_, is_terminal);
-    Config::Utility::validateTerminalFilters(filters[i].name(), "http", is_terminal,
+    auto& factory = processFilter(filters[i], i, "http", filter_factories_, is_terminal);
+    Config::Utility::validateTerminalFilters(filters[i].name(), factory.name(), "http", is_terminal,
```

Comment:
I think it might be cleaner to just move the call to `validateTerminalFilters` inside `processFilter`. That way we don't have to hand back the factory reference and can get rid of the update of `is_terminal` by reference (that var can just be internal to processFilter). What do you think?

Reply:
I agree. I'll work on making that change.